waveanalysis/summarize_save/summarize_images.py

```diff
@@ -0,0 +1,252 @@
+            for channel in range(num_channels):
+                pcnt_no_parameter = np.count_nonzero(np.isnan(measurement[channel])) / measurement[channel].shape[0] * 100
+                param = 'Peaks' if name == 'Peak Amp' else 'Periods'
+                file_data_summary[f'Ch {channel + 1} Pcnt No {param}'] = pcnt_no_parameter
+                for ind, stat in enumerate(stats_location):
+                    file_data_summary[f'Ch {channel + 1} {stat} {name}'] = parameters_with_stats_dict[name][channel][ind + 1]
+        # All parameters that are not wave speed
+        elif name not in ['Wave Speed']:
+            for channel in range(num_channels):        
+                for ind, stat in enumerate(stats_location):
+                    file_data_summary[f'Ch {channel + 1} {stat} {name}'] = parameters_with_stats_dict[name][channel][ind + 1]
+        # Wave Speed is a single value, so it doesn't need to be separated by channel
+        elif name in ['Wave Speed']:
+            for ind, stat in enumerate(stats_location):
+                print(parameters_with_stats_dict[name])
+                file_data_summary[f'{stat} {name}'] = parameters_with_stats_dict[name][0][ind + 1]
+
+    return file_data_summary
+
+def combine_stats_rolling(
+    img_props_dict: dict,
+    img_parameters_dict: dict,
+    indv_ccfs: np.ndarray,
+) -> pd.DataFrame:
+    '''
+    Combine statistics for rolling analysis.
+
+    Args:
+        img_props_dict (dict): A dictionary containing image properties.
+        img_parameters_dict (dict): A dictionary containing image parameters.
+        indv_ccfs (np.ndarray): An array containing individual cross-correlation functions.
+
+    Returns:
+        pd.DataFrame: A DataFrame containing the combined statistics.
+
+    '''
+    # Extract image properties from the dictionary
+    num_channels = img_props_dict['num_channels']
+    num_bins = img_props_dict['num_bins']
+    num_submovies = img_props_dict['num_submovies']
+    channel_combos = img_props_dict['channel_combos']
+
+    # Extract image parameters from the dictionary
+    indv_periods = img_parameters_dict['Period']
+    indv_shifts = img_parameters_dict['Shift']
+    indv_peak_widths = img_parameters_dict['Peak Width']
+    indv_phase_shifts = img_parameters_dict['% Phase Shift']
+
+    # Define the statistics to calculate
+    stat_name_and_func = {'Mean' : np.nanmean,
+                            'Median' : np.nanmedian,
+                            'StdDev' : np.nanstd                            
+                        }
+    
+    # Extract image properties from the dictionary
+    all_submovie_summary = []
+    
+    # Loop through each submovie
+    for submovie in range(num_submovies):
+        # Initialize dictionary to store the summary for the current submovie
+        submovie_summary = {'Submovie': submovie + 1}
+
+        # Calculate percentage of no shifts for each channel combination
+        if num_channels > 1:
```

Comment:
probably easiest to unpack `indv_shifts` and `indv_phase_shifts` inside this block since they won't exist otherwise